apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -69,13 +70,25 @@
 
     if (isUsernameRecovery) {
         // Username Recovery Scenario
-        if (StringUtils.isNotBlank(callback) && !Utils.validateCallbackURL(callback, tenantDomain,
-            IdentityRecoveryConstants.ConnectorConfig.RECOVERY_CALLBACK_REGEX)) {
-            request.setAttribute("error", true);
-            request.setAttribute("errorMsg", IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
-                        "Callback.url.format.invalid"));
-            request.getRequestDispatcher("error.jsp").forward(request, response);
-            return;
+        try {
+            if (StringUtils.isNotBlank(callback) && !Utils.validateCallbackURL(callback, tenantDomain,
+                IdentityRecoveryConstants.ConnectorConfig.RECOVERY_CALLBACK_REGEX)) {
+                request.setAttribute("error", true);
+                request.setAttribute("errorMsg", IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
+                    "Callback.url.format.invalid"));
+                request.getRequestDispatcher("error.jsp").forward(request, response);
+                return;
+            }
+        } catch (IdentityRuntimeException e) {
+            if (StringUtils.isBlank(tenantDomain)) {
```

Comment:
Can we null check `tenantDomain` without waiting for an error to be thrown? So, ideally, we should null check `tenantDomain` before calling `Utils.validateCallbackURL`.